packages/grid/_modules_/grid/hooks/features/virtualization/useGridNoVirtualization.ts

```diff
@@ -69,6 +77,25 @@ export const useGridNoVirtualization = (
     windowRef,
   ]);
 
+  const disableVirtualization = React.useCallback((): void => {
+    setGridState((state) => ({
+      ...state,
+      containerSizes: {
+        ...state.containerSizes!,
+        renderingZone: {
+          ...state.containerSizes!.renderingZone,
+          height: visibleRowCount * props.rowHeight!,
+        },
+        renderingZonePageSize: visibleRowCount,
+      },
+      viewportSizes: {
+        ...state.viewportSizes,
```

Comment:
The width must also be updated to avoid clipping the content during print. You can get the total width occupied by the columns using `gridColumnsMetaSelector`.

https://github.com/mui-org/material-ui-x/blob/50aae6943c4f865d4658e56adda815dca1386d13/packages/grid/_modules_/grid/hooks/features/virtualization/useGridVirtualization.ts#L212

<img width="300" src="https://user-images.githubusercontent.com/42154031/135010435-7d264d9e-c741-44ae-b216-f3724e0ce7b5.png" alt="image" style="max-width: 100%;">

Reply:
Updated - forcing the `width` on the clone directly solves the issue